### PR TITLE
Multiple key fix

### LINKED ISSUE #116: Does not handle multiple keys in a source file
## Affected Puppet, Ruby, OS and module versions/distributions

- Puppet:  PE 2018.1.4 / Puppet agent 5.5.6
- Module version: 3.1.0

## How to reproduce (e.g Puppet code you use)
yum::gpgkey { '/etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-release':
  ensure => present,
  source => 'file:///etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-release',
}

rpm -e gpg-pubkey-2fa658e0
puppet agent -t

## What are you seeing
The key is NOT replaced because the manifest only looks for the first key (head -1)

## What behaviour did you expect instead
All keys in the file should be checked.

### PR DESCRIPTION
**Pull request by linuxdan**
#### Pull Request (PR) description
Added a function to return multiple keys from a key file

#### This Pull Request (PR) fixes the following issues

Fixes #116

### CODE REVIEW

**Comment by bastelfreak** [comment]
Hi @linuxdan, thanks for providing the patch! Can you take a look at the travis issues?

**Comment by linuxdan** [comment]
What issue number(s) ?
I saw no “travis”

**Comment by bastelfreak** [comment]
2018-09-29-235537_723x143_scrot
Travis is our CI platform. You can click on the `Details` button to get to all CI jobs. Some of them failed. Let us know if you've got further questions. We're available in our IRC channel #voxpupuli on freenode and on https://puppetcommunity.slack.com

**Comment by linuxdan** [comment]
manifests/gpgkey.pp:64:variable_scope:WARNING:top-scope variable being used without an explicit namespace

line 64 is 
    $rpmname = "gpg-pubkey-${key}"

What the heck ?

**Comment by linuxdan** [comment]
I'm on Slack in the #forge-modules channel

**Comment by linuxdan** [comment]
manifests/gpgkey.pp:64:variable_scope:WARNING:top-scope variable being used without an explicit namespace
line 64 is now
$the_rpmname = "gpg-pubkey-${kkey}"

What the heck ?

**Comment by linuxdan** [comment]
OK.  Cleaned up what I can.
Not sure why it hates manifests/gpgkey.pp:64
and the other travis errors are out of scope of my changes.

File: manifests/gpgkey.pp [hunk 1]
```diff
@@ -58,25 +58,26 @@
...
+  $keys = yum::get_gpg_keys($path)
 
-  case $ensure {
-    'present', default: {
-      exec { "rpm-import-${name}":
-        path    => '/bin:/usr/bin:/sbin/:/usr/sbin',
-        command => "rpm --import ${path}",
-        unless  => "rpm -q ${rpmname}",
-        require => File[$path],
+  $keys.each |String kkey| {
```
Comment (bastelfreak) [inline manifests/gpgkey.pp]: just for a test, could you remove the keyword `String` here? I'm not sure if the datatype here is a Puppet 5 only feature.

Comment (linuxdan) [reply manifests/gpgkey.pp]: Still getting 
manifests/gpgkey.pp:64:variable_scope:WARNING:top-scope variable being used without an explicit namespace
on first 3 tests

Comment (linuxdan) [reply manifests/gpgkey.pp]: I am out of ideas and tired of tinkering.
@bastelfreak  - Any suggestions ?

**Pull request closed without merging**